catalog/rocks/cataloger.go

```diff
@@ -132,7 +179,58 @@ func (c *cataloger) DeleteBranch(ctx context.Context, repository string, branch
 }
 
 func (c *cataloger) ListBranches(ctx context.Context, repository string, prefix string, limit int, after string) ([]*catalog.Branch, bool, error) {
-	panic("not implemented") // TODO: Implement
+	repositoryID, err := graveler.NewRepositoryID(repository)
+	if err != nil {
+		return nil, false, err
+	}
+	prefixBranch, err := graveler.NewBranchID(prefix)
+	if err != nil {
+		return nil, false, err
+	}
+	afterBranch, err := graveler.NewBranchID(after)
+	if err != nil {
+		return nil, false, err
+	}
+	// normalize limit
+	if limit < 0 || limit > ListBranchesLimitMax {
+		limit = ListBranchesLimitMax
+	}
+	it, err := c.EntryCatalog.ListBranches(ctx, repositoryID)
+	if err != nil {
+		return nil, false, err
+	}
+	if afterBranch == "" {
+		it.SeekGE(prefixBranch)
+	} else {
+		it.SeekGE(afterBranch)
+	}
```

Comment:
what if after < prefix ?

Reply:
My implementation should consider it as a miss and return no items. I can check if after < from the prefix and use prefix as you suggest I agree it will have the more expected behavior.